route/build/src/init_model_data.f90

```diff
@@ -28,9 +28,49 @@ MODULE init_model_data
 public :: init_ntopo_data
 public :: init_state_data
 public :: update_time
+public :: init_pio
 
 CONTAINS
 
+ ! *********************************************************************
+ ! public subroutine: initialize pio system and decomposition
+ ! *********************************************************************
+ SUBROUTINE init_pio(pid, nNodes, ierr, message)
+
+  USE pio_utils,       ONLY: pio_sys_init
+  USE globalData,      ONLY: pioSystem
+  USE globalData,      ONLY: mpicom_route
+  USE globalData,      ONLY: pio_numiotasks
+  USE globalData,      ONLY: pio_rearranger
+  USE globalData,      ONLY: pio_root
+  USE globalData,      ONLY: pio_stride
+  USE globalData,      ONLY: runMode
+  USE pio_decomp_data, ONLY: set_pio_decomp
+
+  implicit none
+  ! Argument variables
+  integer(i4b), intent(in)    :: pid              ! proc id
+  integer(i4b), intent(in)    :: nNodes           ! number of procs
+  integer(i4b), intent(out)   :: ierr             ! error code
+  character(*), intent(out)   :: message          ! error message
+  ! local variables
+  character(len=strLen)       :: cmessage         ! error message from subroutine
+
+  ! pio system initialization
+  if (trim(runMode)=='standalone') then
+    pio_numiotasks = nNodes/pio_stride
```

Comment:
Hi @ekluzek,   just below, I create pioSystem if it is standalone, so this should be actual object instead of pointer?   if it is cesm-coupling,  pioSystem is pointing to the one initialized in the coupler with `shr_pio_getiosys(inst_name)` in RtmMod.f90.

Reply:
This is in #431

Reply:
Allocating `pioSystem` before calling pio_sys_init to define it works.